pytket/binders/circuit/main.cpp

```diff
@@ -31,6 +31,16 @@
 
 namespace py = pybind11;
 
+// A type that should be raised as an exception in Python
+class OpException : public std::exception {
```

Comment:
This feels quite ad-hoc. We do need to improve exception handling in the Python layer but for now we just let exceptions propagate from C++ and I think we should stick with that, rather than do anything special for this function.

Reply:
Does it ? I felt it would serve later as a model for custom exceptions in the bindings that would be specific. But OK I can remove it.

Reply:
I think what we really need is (a) improved exceptions in the C++ and (b) systematically translating these into Python exceptions. I'll make sure we have captured these tasks somewhere.

Reply:
Well, isn't it what this does ?

Reply:
Not exactly. This handles a particular case, but I don't want to have to add special exception-handling code in every pytket method, that would quickly become unmaintainable. Need some generic way to capture all relevant information in the C++ exception and pass it through to Python.

Reply:
OK if you're after genericity. But why not having a central error file that contains all variants and use the `pybind11` error registration machinery to translate them adequately to Python errors ? Just an idea really.